src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/MergeSigningInfo.cs

```diff
@@ -0,0 +1,329 @@
+                    {
+                        new StrongNameSignInfo()
+                        {
+                            CertificateName = "ThisIsACert",
+                            Include = "IncludeMe",
+                            PublicKeyToken = "123456789abcde12"
+                        },
+                        new StrongNameSignInfo()
+                        {
+                            CertificateName = "AnotherCert",
+                            Include = "StrongName",
+                            PublicKeyToken = "123456789abcde12"
+                        }
+                    },
+                    ItemsToSign = new List<ItemsToSign>()
+                };
+
+        [SetUp]
+        public void SetupMergeSigningInfo()
+        {
+            pushMetadata = new PushMetadataToBuildAssetRegistry();
+        }
+
+        [Test]
+        public void GivenCompatibleSigningInfo()
+        {
+            SigningInformation actualMerged = pushMetadata.MergeSigningInfo(ExpectedSigningInfo.Concat(ExpectedSigningInfo2).ToList());
+            SharedMethods.CompareSigningInformation(actualMerged, ExpectedMergedSigningInfo);
+        }
+
+        [Test]
+        public void GivenDuplicateSigningInfo()
+        {
+            SigningInformation actualMerged = pushMetadata.MergeSigningInfo(ExpectedSigningInfo.Concat(ExpectedSigningInfo).ToList());
+            SharedMethods.CompareSigningInformation(actualMerged, ExpectedSigningInfo.First());
+        }
+
+        [Test]
+        public void GivenTwoPartialSigningInfoMetadatas()
+        {
+            Action act = () => pushMetadata.MergeSigningInfo(new List<SigningInformation> { PartialSigningInfo1, PartialSigningInfo2 });
+            act.Should().Throw<Exception>().WithMessage("Can't merge if one or more manifests have different build id, collection URI or project.");    
+        }
+
+        [Test]
+        public void GivenTwoPartialSigningInfosWithEmptySections()
+        {
+            SigningInformation actualMerged = pushMetadata.MergeSigningInfo(new List<SigningInformation> { PartialSigningInfo3, PartialSigningInfo4 });
+            SharedMethods.CompareSigningInformation(actualMerged, MergedPartialSigningInfos);
+        }
+
+        [Test]
+        public void GivenIncompatibleSigningInfos()
+        {
+            Action act = () => pushMetadata.MergeSigningInfo(new List<SigningInformation> { PartialSigningInfo3, IncompatibleSigningInfo });
+            act.Should().Throw<Exception>().WithMessage("Can't merge if one or more manifests have different build id, collection URI or project.");
+        }
+
+        [Test]
+        public void GivenNullSigningInfoList()
+        {
+            Action act = () => pushMetadata.MergeSigningInfo(null);
+            act.Should().Throw<ArgumentNullException>();
+        }
+
+        [Test]
+        public void GivenEmptySigningInfoList()
+        {
+            pushMetadata.MergeSigningInfo(new List<SigningInformation>());
```

Comment:
Is there something we want to assert on this test?

Reply:
That's odd, it used to... Fixed it up.